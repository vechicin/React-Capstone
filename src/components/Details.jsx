import React from 'react';

const Details = () => (
  <p>This is the details page!</p>
);

export default Details;
