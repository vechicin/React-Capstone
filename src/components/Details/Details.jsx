import React from 'react';

const Details = () => (
  <p>This is the first details page!</p>
);

export default Details;
