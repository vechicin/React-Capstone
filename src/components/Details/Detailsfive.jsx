import React from 'react';

const Detailsfive = () => (
  <p>This is the fifth details page!</p>
);

export default Detailsfive;
