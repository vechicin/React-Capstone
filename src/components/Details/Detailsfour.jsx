import React from 'react';

const Detailsfour = () => (
  <p>This is the fourth details page!</p>
);

export default Detailsfour;
