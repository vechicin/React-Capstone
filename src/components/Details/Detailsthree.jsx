import React from 'react';

const Detailsthree = () => (
  <p>This is the third details page!</p>
);

export default Detailsthree;
