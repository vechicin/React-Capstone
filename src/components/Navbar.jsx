import React from 'react';
import { NavLink } from 'react-router-dom';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
  ];

  return (
    <nav>
      <ul className="navbar-ul">
        {links.map((link) => (
          <li className="d-flex" key={link.id}>
            <NavLink className="btn btn-light" to={link.path} />
            <img src="apple.svg" alt="apple logo" />
            <h1>Financial Statements</h1>
            <img src="mic-fill.svg" alt="microphone" />
            <img src="gear-fill.svg" alt="gear" />
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
