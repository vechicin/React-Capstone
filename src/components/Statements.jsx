import React from 'react';
import { useSelector } from 'react-redux';

const Statements = () => {
  const statement = useSelector((state) => state.home.data);
  console.log(statement);

  return (
    <div>
      {/* <li>{date}</li>
      <li>{currency}</li> */}
    </div>
  );
};

export default Statements;
