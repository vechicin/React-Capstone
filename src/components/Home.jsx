import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { fetchData } from '../redux/home/home';
import Details from './Details';

const Home = () => {
  const data = useSelector((state) => state.home.data);
  const dispatch = useDispatch();

  useEffect(() => {
    if (data.length === 0) {
      dispatch(fetchData());
    }
  }, []);

  return (
    <div>
      {data.map((item) => (
        <NavLink key={item.year} to={`/details/${item.date}`}>
          <Details year={item.year} />
        </NavLink>
      ))}
    </div>
  );
};

export default Home;
