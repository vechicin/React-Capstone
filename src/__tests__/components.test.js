import { render } from "@testing-library/react"
import { Provider } from "react-redux"
import { Router } from "react-router"
import { PersistGate } from "redux-persist/integration/react"
import Details from "../components/Details"
import Home from "../components/Home"
import Navbar from "../components/Navbar"
import Statements from "../components/Statements"
import store, { persistor } from "../redux/configureStore"

describe('Components testing', () => {
  test('Home component', () => {
    const tree = render(<Provider store={store}><Home /></Provider>);
    expect(tree).toMatchSnapshot();
  });

  test('Details component', () => {
    const tree = render(<Provider store={store}><Details /></Provider>);
    expect(tree).toMatchSnapshot();
  });

  test('Statements component', () => {
    const tree = render(<Provider store={store}><Statements /></Provider>);
    expect(tree).toMatchSnapshot();
  });
});
