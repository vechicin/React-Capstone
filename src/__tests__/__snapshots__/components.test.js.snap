// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components testing Details component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="home-ul"
      >
        <li
          class="list-group-item"
        />
        <li
          class="list-group-item"
        />
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="home-ul"
    >
      <li
        class="list-group-item"
      />
      <li
        class="list-group-item"
      />
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Components testing Home component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="list-group"
      >
        <input
          placeholder="Search by year..."
          type="search"
        />
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="list-group"
    >
      <input
        placeholder="Search by year..."
        type="search"
      />
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
